src/main/java/lab2/Lab2.java

```diff
@@ -0,0 +1,199 @@
+*/
+import java.io.InputStream;
+/*package: java.net
+purpose: used for networking, such as working with URLs, sockets, and internet addresses.
+classes:
+URI, URL – represent web addresses
+Socket and ServerSocket – for network communication
+InetAddress and URLConnection
+ */
+import java.net.URI;
+/*
+kinds of Classes:
+HttpClient – sends HTTP requests
+HttpRequest – builds HTTP requests
+HttpResponse – handles responses
+HttpRequest.BodyPublisher, HttpResponse.BodyHandler - control body types
+
+ */
+import java.net.http.HttpClient;
+/*
+purpose: represents the HTTP client. It is used to send HTTP requests and receive responses.
+constructor: HttpClient.newHttpClient() to create an instance (factory method)
+classes:
+.send(request, handler) -	Sends a synchronous request
+.sendAsync(request, handler)- Sends an asynchronous request
+ */
+
+import java.net.http.HttpRequest;
+
+/*
+purpose: represents an HTTP response from a server. It contains status code, headers, and body.
+classes:
+
+HttpClient - Main class used to send requests and receive responses. Can be configured with options like redirect policy, proxy, and SSL.
+HttpRequest	- Represents an HTTP request. Built using a builder pattern (HttpRequest.newBuilder()).
+HttpResponse<T>	- Represents an HTTP response, with generic type T for the response body (e.g., String, InputStream).
+HttpRequest.BodyPublishers - Contains static methods to create request bodies (e.g., ofString(), ofFile()).
+HttpResponse.BodyHandlers -	Contains static methods to process response bodies (e.g., ofString(), ofInputStream()).
+WebSocket - Provides support for WebSocket communication (real-time, full-duplex).
+HttpHeaders - Represents HTTP headers (used in requests and responses).
+HttpTimeoutException - Exception thrown when a request times out.
+ */
+import java.net.http.HttpResponse;
+
+
+//method: AvatarGenerator
+//type: user-defined class
+//purpose: generate and display random avatars from the internet using the DiceBear API.
+
+void main() {
+    try {
+        var avatarStream = getRandomAvatarStream();
+        showAvatar(avatarStream);
+    } catch (IOException | InterruptedException e) {
+        e.printStackTrace();
+    }
+
+}
+
+//method: getRandomAvatarStream
+//type: public static
+//throws a IOException and InterruptedException when encountered certain runtime problems.
+InputStream getRandomAvatarStream() throws IOException, InterruptedException {
+
+
+    // Pick a random style
+    //Variable type: String[]
+    //list of styles
+    String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile", "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei", "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" };
+    var style = styles[(int)(Math.random() * styles.length)];
+    //method: Math.random()
```

Comment:
You've done a good job of identifying the type and purpose of each variable here, but I don't see any information about if they are primitive/reference